### ISSUE TITLE
Transformation Item Antidotes do not work

### ISSUE THREAD

**Issue by beffymaroo** [HabitRPG/habitica#10160]
I'm seeing reports of this in both staging and production, and so far it's been confirmed for at least Shiny Seeds and Spooky Sparkles.

When one goes to purchase an antidote, one is able to buy multiples which may not be desired behavior since those items were instant use. Upon purchasing any amount of them, no gold is deducted and my avatar is not un-transformed. This persists upon refresh or reload.

**Comment by Alys** [comment]
I've marked this important priority because a Transformation item will be re-released on 3 April so we'd ideally want to have this fixed before then.

**Comment by Alys** [comment]
Console log from staging:
```
Unhandled promise rejection Object { name: "t", httpCode: 404, 
message: "Item "petalFreePotion" not found.", stack: "" }
```

**Comment by Cappycot** [comment]
The purchase of these antidotes is being treated as if it were a commodity such as a hatching potion or egg while it should behave like an instant-use health potion.

Caleb, who's currently working on issue #9758, is likely to be fixing this issue as well since we found that the two of these issues are closely tied together.

**Comment by caleb-fung** [comment]
@Alys I believe I found a fix to this issue and #9758.

**Comment by caleb-fung** [comment]
I submitted a pull request for this issue because it is similar to #9758. Although my code failed the Coverage test, it passed all Travis tests. What other tests should I create if needed?
